dev-libs/poco/poco-1.12.2.ebuild

```diff
@@ -0,0 +1,144 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="C++ libraries for building network-based applications"
+HOMEPAGE="https://pocoproject.org/"
+SRC_URI="https://github.com/pocoproject/${PN}/archive/${P}-release.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN}-${P}-release"
+
+LICENSE="Boost-1.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 x86"
+IUSE="7z activerecord cppparser +crypto +data examples +file2pagecompiler iodbc +json mariadb +mongodb mysql +net odbc +pagecompiler pdf pocodoc postgres prometheus sqlite +ssl test +util +xml +zip"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	7z? ( xml )
+	file2pagecompiler? ( pagecompiler )
+	iodbc? ( odbc )
+	mongodb? ( data )
+	mysql? ( data )
+	odbc? ( data )
+	postgres? ( data )
+	pagecompiler? ( json net util xml )
+	pocodoc? ( cppparser util xml )
+	sqlite? ( data )
+	ssl? ( util )
+	test? ( data? ( sqlite ) json util xml )
+"
+
+BDEPEND="virtual/pkgconfig"
+RDEPEND="
+	>=dev-libs/libpcre-8.42
+	activerecord? ( !app-arch/arc )
+	mysql? ( !mariadb? ( dev-db/mysql-connector-c:0= )
+	mariadb? ( dev-db/mariadb-connector-c:0= ) )
+	postgres? ( dev-db/postgresql:= )
+	odbc? (
+		iodbc? ( dev-db/libiodbc )
+		!iodbc? ( dev-db/unixODBC )
+	)
+	sqlite? ( dev-db/sqlite:3 )
+	ssl? (
+		dev-libs/openssl:0=
+	)
+	xml? ( dev-libs/expat )
+	zip? ( sys-libs/zlib )
+"
+DEPEND="${RDEPEND}"
+
+PATCHES=( "${FILESDIR}/${PN}-1.10.1-iodbc-incdir.patch" )
+
+src_prepare() {
+	cmake_src_prepare
+
+	if use test ; then
+		# ignore missing tests on experimental library
+		# and tests requiring running DB-servers, internet connections, etc.
+		sed -i -e '/testsuite/d' \
+			{Data/{MySQL,ODBC},MongoDB,Net,NetSSL_OpenSSL,PDF,Redis}/CMakeLists.txt || die
+		# Poco expands ~ using passwd, which does not match $HOME in the build environment
+		sed -i -e '/CppUnit_addTest.*testExpand/d' \
+			Foundation/testsuite/src/PathTest.cpp || die
+		# ignore failing Crypto test since upstream does not seem to care,
+		# see https://github.com/pocoproject/poco/issues/1209
+		sed -i -e '/RSATest, testRSACipherLarge/d' \
+			Crypto/testsuite/src/RSATest.cpp || die
+	fi
+
+	if use mariadb ; then
+		# Fix MariaDB detection
+		sed -i -e 's~/usr/include/mysql~~' \
+			-e 's/STATUS "Couldn/FATAL_ERROR "Couldn/' \
+		cmake/FindMySQL.cmake || die
+	else
+		# Fix MySQL detection
+		sed -i -e 's/mysqlclient_r/mysqlclient/' \
+			-e 's/STATUS "Couldn/FATAL_ERROR "Couldn/' \
+		cmake/FindMySQL.cmake || die
+	fi
```

Comment:
I don't think this works, if you try to build it with `+mysql +mariadb` it will error out with:
```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MySQL (missing: MYSQL_INCLUDE_DIR MYSQL_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindMySQL.cmake:107 (find_package_handle_standard_args)
  CMakeLists.txt:125 (find_package)
```

Reply:
Thanks! It should be fixed now.